engine/table/src/main/java/io/deephaven/stream/StreamPublisher.java

```diff
@@ -30,4 +32,10 @@ public interface StreamPublisher {
      * {@link StreamConsumer#accept(WritableChunk[]) accept} method.
      */
     void flush();
+
+    /**
+     * Shutdown this StreamPublisher. Implementations should stop publishing new data and release any related resources
+     * as soon as practicable.
+     */
+    void shutdown();
```

Comment:
I like this functionality. That said, is there a reason to prefer "shutdown" over "close()" / AutoCloseable?

Reply:
The goal here was just to move shutdown ownership up a layer, so we don't have to have every consumer implement a `setShutdownCallback` method. I don't really want it called externally, and I prefer to save `AutoCloseable` or `SafeCloseable` implementations for resource cases.